httpgrpc/protocol_versions.go

```diff
@@ -30,3 +36,24 @@ const (
 	UnaryRpcContentType_V1  = "application/x-protobuf"
 	StreamRpcContentType_V1 = "application/x-httpgrpc-proto+v1"
 )
+
+func getCodec(contentType string) encoding.Codec {
+	if contentType == UnaryRpcContentType_V1 || contentType == StreamRpcContentType_V1 {
+		return encoding.GetCodec(grpcproto.Name)
+	}
+
+	// Try to find the best encoder.
+	if strings.HasPrefix(contentType, "application/grpc-") {
```

Comment:
Thank you for removing the grpc-web stuff. But I think the same argument goes for this one, too: the format on the wire we are using is not compatible with the actual grpc format. It uses a slightly different envelope, to support optional per-message compression, and also requires trailers for communicating status codes.

Reply:
I'm confused tho-- that stuff is in our own repo.  This simply allows anyone to plug into the open source lib.  In theory, someone could plug in a codec that is compatible with grpc right?

Reply:
No, the actual grpc content types have message framing that is different from what this implementation does. The codec only influences the message contents, not the envelope. The actual grpc protocol also does _not_ include the status code, error details, or trailers as part of the content; they are conveyed via HTTP trailers.